src/serverenvironment.cpp

```diff
@@ -1542,43 +1481,11 @@ void ServerEnvironment::getAddedActiveObjects(PlayerSAO *playersao, s16 radius,
 	f32 radius_f = radius * BS;
 	f32 player_radius_f = player_radius * BS;
 
-	if (player_radius_f < 0)
-		player_radius_f = 0;
-	/*
-		Go through the object list,
-		- discard removed/deactivated objects,
-		- discard objects that are too far away,
-		- discard objects that are found in current_objects.
-		- add remaining objects to added_objects
-	*/
-	for (auto &ao_it : m_active_objects) {
-		u16 id = ao_it.first;
-
-		// Get object
-		ServerActiveObject *object = ao_it.second;
-		if (object == NULL)
-			continue;
+	if (player_radius_f < 0.0f)
+		player_radius_f = 0.0f;
 
-		if (object->isGone())
-			continue;
-
-		f32 distance_f = object->getBasePosition().
-			getDistanceFrom(playersao->getBasePosition());
-		if (object->getType() == ACTIVEOBJECT_TYPE_PLAYER) {
-			// Discard if too far
-			if (distance_f > player_radius_f && player_radius_f != 0)
-				continue;
-		} else if (distance_f > radius_f)
-			continue;
-
-		// Discard if already on current_objects
-		std::set<u16>::iterator n;
-		n = current_objects.find(id);
-		if(n != current_objects.end())
-			continue;
-		// Add to added_objects
-		added_objects.push(id);
-	}
+	m_ao_manager.getAddedActiveObjectsAroundPlayer(playersao->getBasePosition(), radius_f,
```

Comment:
IMO this function is too specific for the ActiveObject class. Suggestion: Add an iterator so that the (real active) active objects can read out quickly.

Reply:
i wanted to prevent this to make the active object really owned by manager and protect them. it's also why there is a ServerAO Manager :) and i should rename to getAddedActiveObjectsAroundPosition because before we passed the playerSAO and now it's just a position.

Reply:
What's the profit of forwarding all functions to the manager? Just speaking of the code, it will be more independent - but also gets a complexer structure. The two classes cannot become independent anyway.